plugins/trace_api_plugin/request_handler.cpp

```diff
@@ -62,6 +62,42 @@ namespace {
 
    }
 
+   fc::variants process_actions(const std::vector<action_trace_v1>& actions, const data_handler_function_v1& data_handler, const yield_function& yield ) {
+      fc::variants result;
+      result.reserve(actions.size());
+
+      // create a vector of indices to sort based on actions to avoid copies
+      std::vector<int> indices(actions.size());
+      std::iota(indices.begin(), indices.end(), 0);
+      std::sort(indices.begin(), indices.end(), [&actions](const int& lhs, const int& rhs) -> bool {
+         return actions.at(lhs).global_sequence < actions.at(rhs).global_sequence;
+      });
+
+      for ( int index : indices) {
+         yield();
+
+         const auto& a = actions.at(index);
+         auto action_variant = fc::mutable_variant_object()
+               ("global_sequence", a.global_sequence)
+               ("receiver", a.receiver.to_string())
+               ("account", a.account.to_string())
+               ("action", a.action.to_string())
+               ("authorization", process_authorizations(a.authorization, yield))
+               ("data", fc::to_hex(a.data.data(), a.data.size()))
+               ("return_value", fc::to_hex(a.return_value.data(), a.return_value.size()));
```

Comment:
"return_value" needs to be expanded like "params"

Reply:
Expnaded.